src/test/java/test/suites/github1850/DuplicateChildSuitesInitializationTest.java

```diff
@@ -0,0 +1,76 @@
+package test.suites.github1850;
+
+import org.testng.annotations.Test;
+import org.testng.xml.Parser;
+import org.testng.xml.XmlSuite;
+import test.SimpleBaseTest;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * This test checks that TestNG can handle duplicate child suites when we have the following set of files:
+ *
+ * - parent-suite-with-duplicate-child -> [child-suite-1, children/child-suite-3, child-suite1, children/child-suite-3]
+ * - children/child-suite-3 -> [../child-suite-2, child-suite-4, morechildren/child-suite-5]
+ *
+ * SHOULD return a XmlSuite object with following structure:
+ *
+ * parent-suite-with-duplicate-child
+ * 	├───child-suite-1
+ * 	├───child-suite-3
+ * 	│   ├───child-suite-2
+ * 	│   ├───child-suite-4
+ * 	│   └───child-suite-5
+ * 	├───child-suite-1(0)
+ * 	└───child-suite-3(0)
+ * 		├───child-suite-2(0)
+ * 		├───child-suite-4(0)
+ * 		└───child-suite-5(0)
+ *
+ * but NOT like:
+ *
+ * parent-suite-with-duplicate-child
+ * 	├───child-suite-1
+ * 	├───child-suite-3
+ * 	├───child-suite-1(0)
+ * 	└───child-suite-3(0)
+ * 		├───child-suite-2
+ * 		├───child-suite-4
+ * 		├─-─child-suite-5
+ * 		├───child-suite-2(0)
+ * 	    ├───child-suite-4(0)
+ * 	    └───child-suite-5(0)
+ *
+ * Check the <code>checksuitesinitialization</code> folder under test resources
+ *
+ * @author Chao Qin
+ */
+public class DuplicateChildSuitesInitializationTest extends SimpleBaseTest {
+
+
+
+ /**
+   * Parent suite contains duplicate child suites
+   */
+  @Test
+  public void checkDuplicateChildSuites() throws IOException {
+    String path = getPathToResource("checksuitesinitialization/parent-suite-with-duplicate-child.xml");
+    Parser parser = new Parser(path);
+    List<XmlSuite> suites = parser.parseToList();
+      XmlSuite rootSuite = suites.get(0);
```

Comment:
Formatting issue. Can you please have this method reformatted so that its indented properly.